src/sentry/sentry_metrics/querying/samples_list.py

```diff
@@ -84,10 +84,62 @@ def get_spans_by_key(self, span_ids: list[tuple[str, str, str]]):
         return builder.process_results(query_results)
 
 
+class SegmentsSamplesListExecutor(SamplesListExecutor):
+    @classmethod
+    def mri_to_column(cls, mri) -> str | None:
+        if mri == TransactionMRI.DURATION.value:
+            return "duration"
+        return None
+
+    @classmethod
+    def supports(cls, mri: str) -> bool:
+        return cls.mri_to_column(mri) is not None
+
+    def execute(self, offset, limit):
+        span_keys = self.get_span_keys(offset, limit)
+        return self.get_spans_by_key(span_keys)
+
+    def get_span_keys(self, offset: int, limit: int) -> list[tuple[str, str, str]]:
+        rounded_timestamp = f"rounded_timestamp({self.rollup})"
+
+        builder = QueryBuilder(
+            Dataset.Transactions,
```

Comment:
Are we currently writing span segments to both indexed transactions and indexed spans?

Reply:
To the best of my understanding, a segment is just a transaction. And there's already talk of making sure we write all segments to both transactions and indexed spans tables for the medium term. Long term this may change but it's easy enough to rip out this query builder and swap it for the spans query builder.